src/core_plugins/kibana/public/dashboard/lib/filter_utils.js

```diff
@@ -0,0 +1,51 @@
+import _ from 'lodash';
+
+/**
+ * @typedef {Object} QueryFilter
+ * @property query_string {Object}
+ * @property query_string.query {String}
+ */
+
+export class FilterUtils {
+  /**
+   *
+   * @param filter
+   * @returns {Boolean} True if the filter is of the special query type
+   * (e.g. goes in the query input bar), false otherwise (e.g. is in the filter bar).
+   */
+  static isQueryFilter(filter) {
+    return filter.query && filter.query.query_string && !filter.meta;
+  }
+
+  /**
+   *
+   * @param {SavedDashboard} dashboard
+   * @returns {Array.<Object>} An array of filters stored with the dashboard. Includes
+   * both query filters and filter bar filters.
+   */
+  static getDashboardFilters(dashboard) {
```

Comment:
I find `getSearchSourceFilters` pretty confusing but maybe thats because I'm still not sure what a search source is. :)

Going to leave as is, as I'm hoping since this is a static function that only takes in the dashboard, it's obvious it doesn't have to do with the state on scope.  In View/Edit mode I will be making a distinction between initial dashboard filters, and current filters, so hopefully that will also improve clarity.